aten/src/ATen/test/verify_api_visibility.cpp

```diff
@@ -0,0 +1,7 @@
+#include <ATen/ATen.h>
+
+#ifdef AT_CUDA_ENABLED
+#error "AT_CUDA_ENABLED should not be visible in public headers"
+#endif
```

Comment:
Can you add a test for `AT_CUDNN_ENABLED` too? I think we are already clean but it would be good to stay that way :)